### PR TITLE
xtimer: Malo xtimer fixes

### PR DESCRIPTION
**Pull request by malosek**
As discussed on the mailing list
https://lists.riot-os.org/pipermail/devel/2016-February/003508.html

wbr
malo

### CODE REVIEW

**Comment by OlegHahm** [comment]
Cool! Thanks for the PR. Would you mind prefixing the title of the PR and the commit messages with `xtimer: `?

**Comment by malosek** [comment]
Better now?

wbr
malo

**Comment by OlegHahm** [comment]
Perfect. The change itself seems also be fine to me, but I would like @kaspar030 or @gebart to confirm this.

File: sys/xtimer/xtimer_core.c [hunk 1]
```diff
@@ -179,7 +179,7 @@ int _xtimer_set_absolute(xtimer_t *timer, uint32_t target)
     }
 
     unsigned state = disableIRQ();
-    if ( !_this_high_period(target) ) {
+    if ( timer->long_target > _long_cnt || !_this_high_period(target) ) {
```
Comment (gebart) [inline sys/xtimer/xtimer_core.c]: could you add some parentheses around each individual condition for better readability?

File: sys/xtimer/xtimer_core.c [hunk 2]
```diff
@@ -217,8 +217,9 @@ static void _add_timer_to_list(xtimer_t **list_head, xtimer_t *timer)
 static void _add_timer_to_long_list(xtimer_t **list_head, xtimer_t *timer)
 {
     while (*list_head
-            && (*list_head)->long_target <= timer->long_target
-            && (*list_head)->target <= timer->target) {
+            && (((*list_head)->long_target < timer->long_target)
+            || ((*list_head)->long_target == timer->long_target
```
Comment (gebart) [inline sys/xtimer/xtimer_core.c]: could you add some parentheses around each individual condition?
i.e. 
```c
     while (*list_head
            && (((*list_head)->long_target < timer->long_target)
            || (((*list_head)->long_target == timer->long_target)
            && ((*list_head)->target <= timer->target)))) {
```

**Comment by gebart** [comment]
I agree with the changes, but could you add some parentheses just to make the code extra clear?

OK to squash the parenthesis fixes into the original commits immediately.

ACK after this is done.

**Pull request merged**